universal-application-tool-0.0.1/app/auth/UATProfile.java

```diff
@@ -1,11 +1,46 @@
 package auth;
 
+import com.google.common.base.Preconditions;
+import java.time.Clock;
+import models.Account;
+import models.Applicant;
 import org.pac4j.core.profile.CommonProfile;
 
 public class UATProfile extends CommonProfile {
+  private final Clock clock;
+  private Applicant applicant;
+
   public UATProfile() {
+    this(Clock.systemDefaultZone());
+  }
+
+  public UATProfile(Clock clock) {
     super();
-    // TODO: write this to the database, read the ID back, and use that as the ID here.
-    setId("TODO");
+    this.clock = clock;
+  }
+
+  public Applicant getApplicant() {
+    if (applicant == null) {
+      this.applicant = new Applicant();
+      this.applicant.id = Long.valueOf(this.getId());
+      this.applicant.refresh();
+    }
+    return applicant;
+  }
+
+  public Account getAccount() {
+    return getApplicant().getAccount();
+  }
+
+  public void init() {
+    Account acc = new Account();
+    acc.save();
+    Applicant newA = new Applicant();
+    newA.getApplicantData().put("$.metadata", "created_time", clock.instant().toString());
```

Comment:
Issue (blocking): we intend to deprecate the generic `put` and `read` methods on `ApplicantData` for better encapsulation in favor of more focused, less generic methods for accessing the underlying JSON data.

Suggestion: add a `setCreateTime` method to `ApplicantData`.

Reply:
Sure thing, done.